lib/intercom/client.rb

```diff
@@ -156,7 +156,7 @@ def execute_request(request)
       request.handle_rate_limit = handle_rate_limit
       request.execute(@base_url, token: @token, api_version: @api_version, **timeouts)
     ensure
-      @rate_limit_details = request.rate_limit_details
+      @rate_limit_details = request.rate_limit_details unless request.rate_limit_details.empty?
```

Comment:
Do you mean?
```suggestion
      @rate_limit_details = request.rate_limit_details unless request.rate_limit_details.nil?
```
I think this is your intention from what I understand of the PR description

Reply:
if its an empty hash we can set it, does not matter. As long as its not nil.